### ISSUE TITLE
Enforce char limits on free text fields in profiles (2000 chars)

### ISSUE THREAD

**Issue by michaltrzesimiech** [rtcharity/eahub.org#724]

**Comment by sebgrebe** [comment]
@taymonbeal I'm thinking about implementing this by simply adding a max_length of 2000 to the relevant fields in the profile model. If I understand [the documentation](https://docs.djangoproject.com/en/2.2/ref/models/fields/#django.db.models.TextField) correctly, this shouldn't have any impact on already existing entries in the database. When I tested this locally, a previous entry of more than 2000 characters was still fully displayed.

Does this sound OK to you?

**Comment by taymonbeal** [comment]
Yes, but this won't actually fully prevent abuse, i.e., a request can still be submitted that's over the limit. It does introduce a speed bump though.

**Comment by sebgrebe** [comment]
> a request can still be submitted that's over the limit

After I had implemented this locally (without any additional frontend work), I was not able to enter anything longer than 2000 characters. Or do you mean something else?

**Comment by taymonbeal** [comment]
This restriction isn't enforced on the server side, so a user can override it in their browser or just send a request some other way.

**Comment by sebgrebe** [comment]
Is there a way to enforce this server side that doesn't break the existing database?

If not, I guess adding this "speed bump" is better than nothing. I don't think we're concerned about abuse, but rather want to give users some guidance on how long their input should be

**Comment by taymonbeal** [comment]
You want to use https://docs.djangoproject.com/en/2.2/ref/validators/#maxlengthvalidator.

**Comment by sebgrebe** [comment]
Changed it. 

What's not ideal about this is that with this validation, the user only sees that there is a max length once they submit their change. 
![Screen Shot 2019-08-23 at 17 55 05](https://user-images.githubusercontent.com/15908208/63609513-29b2da00-c5cf-11e9-84c4-9e30f78c673c.png)

So I've also added a maxlength to the html textarea element, so that the users cannot even enter more than 2000 characters. Less frustrating for the users than being able to submit and then seeing an error message
